src/service/metrics_middleware.ts

```diff
@@ -39,6 +39,29 @@ export const metricsMiddleware = (config: MetricsConfig, log: Logger) => {
     opts.promClient.collectDefaultMetrics = {};
   }
 
-  const bundle = promBundle(opts);
-  return bundle;
+  const metricsMiddleware = promBundle(opts);
+  app.use(metricsMiddleware);
+
+  const httpRequestsInFlight = new promClient.Gauge({
+    name: 'grafana_image_renderer_http_request_in_flight',
+    help: 'A gauge of requests currently being served by the image renderer.',
+  });
+  app.use(requestsInFlightMiddleware(httpRequestsInFlight, excludeRegExp));
+};
+
+const requestsInFlightMiddleware = (httpRequestsInFlight: promClient.Gauge, excludeRegExp: RegExp) => {
+  return (req, res, next) => {
+    const path = req.originalUrl || req.url;
+    if (path.match(excludeRegExp)) {
+      return next();
+    }
+
+    httpRequestsInFlight.inc();
+    onFinished(res, () => {
+      httpRequestsInFlight.dec();
+      next();
```

Comment:
should `next()` really be called here? I suspect that the `next()` call on 65 would be enough to progress. But it was a long time since I wrote nodejs code :)

Reply:
Yes you're right, it's a mistake, I'll remove it.